crates/matrix-sdk-base/src/read_receipts.rs

```diff
@@ -276,8 +277,8 @@ impl RoomReadReceipts {
 
     /// Checks whether the given `TimelineEvent` is part of a thread and should
     /// be ignored.
-    fn should_ignore(event: &TimelineEvent) -> bool {
-        return extract_thread_root(&event.raw()).is_some();
+    fn should_ignore(&self, event: &TimelineEvent, threads_enabled: bool) -> bool {
+        threads_enabled && extract_thread_root(event.raw()).is_some()
     }
```

Comment:
We should probably inline this function (and the comment) into its one caller; the name doesn't say anything about thread, and makes sense only after reading the comment.